plugins/badges-backend/src/badges.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ENTITY_DEFAULT_NAMESPACE } from '@backstage/catalog-model';
+import { Badge, BadgeContext, BadgeFactories } from './types';
+
+function appTitle(context: BadgeContext): string {
+  return context.config.getString('app.title') || 'Backstage';
+}
+
+function entityUrl(context: BadgeContext): string {
+  const e = context.entity!;
+  const entityUri = `${e.kind}/${
+    e.metadata.namespace || ENTITY_DEFAULT_NAMESPACE
+  }/${e.metadata.name}`;
+  const catalogUrl = `${context.config.getString('app.baseUrl')}/catalog`;
+  return `${catalogUrl}/${entityUri}`;
+}
+
+export const createDefaultBadgeFactories = (): BadgeFactories => ({
+  pingback: {
+    createBadge: (context: BadgeContext): Badge | null => {
```

Comment:
haha, ok, I'm not religious about it either, can actually see the point using undefined rather than null, as js can treat them differently, so a null could bite you later.. I'll change it to undefined (just more to type than null :P ).
I was considering having a type that was defined as an optional `Badge`, maybe that would look cleaner too..?
like:
```ts
type OptionalBadge = Badge | undefined;
...
createBadge: (context: BadgeContext): OptionalBadge => { ... }

```

Reply:
Per later comment, I'll refactor the factory function to either return a `Badge` or throw, so we get rid of the optional `undefined` result.